riff-raff/app/controllers/DeployController.scala

```diff
@@ -281,4 +285,31 @@ class DeployController(deployments: Deployments, prismLookup: PrismLookup, deplo
     }
   }
 
+  def deployFiles(projectName: String, id: String) = AuthAction { implicit request =>
+    def pathAndContent(artifact: S3Artifact): Option[(S3Artifact, S3Path, String)] = {
```

Comment:
If this is worth extracting, is it worth testing? If it's useful "logic" then it might be better placed out of the controller with a unit test on it.